### PR TITLE
MINOR: Update `config/producer.properties` to have new producer properties

### PR DESCRIPTION
**Pull request by ijuma**
Also include some trivial clean-ups in `ProducerConfig`and `BaseProducer`.

### CODE REVIEW

**Comment by ijuma** [comment]
@ewencp Could you please review this? The new properties are not the same as the old ones (as often there is no equivalent), but I tried to include similar ones. Let me know if you think we should be including different properties there.

File: config/producer.properties [hunk 1]
```diff
@@ -18,36 +18,28 @@
...
 
-# specifies whether the messages are sent asynchronously (async) or synchronously (sync)
-producer.type=sync
-
-# specify the compression codec for all data generated: none, gzip, snappy, lz4.
-# the old config values work as well: 0, 1, 2, 3 for none, gzip, snappy, lz4, respectively
-compression.codec=none
-
-# message encoder
-serializer.class=kafka.serializer.DefaultEncoder
```
Comment (gwenshap) [inline config/producer.properties:34]: I think this does have an equivalent (ByteSerializer)

File: config/producer.properties [hunk 2]
```diff
@@ -18,36 +18,28 @@
...
+# the maximum size of a request in bytes
+#max.request.size=
 
-# Timeout for event enqueue:
-# 0: events will be enqueued immediately or dropped if the queue is full
-# -ve: enqueue will block indefinitely if the queue is full
-# +ve: enqueue will block up to this many milliseconds if the queue is full
-#queue.enqueue.timeout.ms=
+# the default batch size in bytes when batching multiple records sent to a partition
+#batch.size=
```
Comment (gwenshap) [inline config/producer.properties]: Maybe add linger.ms too? There are usually tuned together... Although TBH, I'm not sure where / why this example file will be used...

File: config/producer.properties [hunk 3]
```diff
@@ -18,36 +18,28 @@
...
-############################# Async Producer #############################
-# maximum time, in milliseconds, for buffering data on the producer queue 
-#queue.buffering.max.ms=
+# the producer will wait for up to the given delay to allow other records to be sent so that the sends can be batched together
+#linger.ms=
 
-# the maximum size of the blocking queue for buffering on the producer 
-#queue.buffering.max.messages=
+# the maximum size of a request in bytes
+#max.request.size=
```
Comment (gwenshap) [inline config/producer.properties:39]: Add request.timeout.ms here too?

**Comment by gwenshap** [comment]
LGTM. Had some minor comments

File: config/producer.properties (hunk 3 above)
Comment (ijuma) [inline config/producer.properties:39]: It's a few lines above, not sure if you saw it.

File: config/producer.properties (hunk 1 above)
Comment (ijuma) [inline config/producer.properties:34]: There are key.serializer and value.serializer but they're also constructor parameters, so I wasn't sure if it was useful to have them here. You think we should have them here?

**Pull request closed without merging**